internal/transport/http2_server.go

```diff
@@ -388,16 +391,19 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 	}
 
 	if !isGRPC {
-		t.controlBuf.put(&headerFrame{
-			streamID: streamID,
-			hf: []hpack.HeaderField{
-				// http status 415: unsupported media type
-				{Name: ":status", Value: "415"},
-				{Name: "grpc-message", Value: encodeGrpcMessage("unsupported media type")},
-				{Name: "grpc-status", Value: strconv.Itoa(int(codes.Internal))},
-				{Name: "content-type", Value: "application/grpc"},
-			},
-			endStream: true,
+		h, err := json.Marshal(hpack.HeaderField{
+			Name: "grpc-message",
+			Value: encodeGrpcMessage("unsupported media type"),
+		})
+		if err != nil {
+			return false
+		}
+
+		t.controlBuf.put(&dataFrame{
+			streamID:    streamID,
+			endStream:   true,
+			h:           h,
+			d:           []byte(fmt.Sprintf("unsupported content-type %q", contentType)),
```

Comment:
I'm not sure if this is right or not. I couldn't find any examples of building a `dataFrame`.

Reply:
How about this: stand up a gRPC server with this change (e.g. run `examples/helloworld/greeter_server`) and use `curl` on it and see if it's able to show the error properly?

Reply:
I've tested a few different responses and `curl` gets an error:
```
curl: (56) Recv failure: Connection reset by peer
@%  
```

I've also tested a flag to set http2:
```
$ curl --http2-prior-knowledge http://localhost:50051
curl: (56) Recv failure: Connection reset by peer
``` 

It doesn't appear to send a readable response back.

Reply:
I don't think you're setting `content-type` to `text/plain` though?

Reply:
The data frame doesn't appear to have a way to set the content type?

Reply:
`content-type` needs to be set in the initial headers.

Reply:
Friendly ping - are you still planning to come back to this?

Reply:
Hey Doug, I've been working on this on and off, and I can't seem to get a plain-text response. I've set the initial headers and attempted to send a data frame. It's very possible I'm building the data incorrectly or something. If you have any suggestions that would be great. Otherwise, I'm probably going to drop this for now.